presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -549,6 +569,34 @@ public void removePreparedStatement(String key)
         deallocatedPreparedStatements.add(key);
     }
 
+    public void addSessionFunction(BasicSignature signature, SqlInvokedFunction function)
+    {
+        requireNonNull(signature, "signature is null");
+        requireNonNull(function, "function is null");
+
```

Comment:
Unnecessary. If null, line 577 will throw.

Reply:
I don't think it would necessarily throw here? This is defensive to make sure we never put nulls as keys or values in our map.